internal/upload/upload.go

```
@@ -0,0 +1,622 @@
+		return nil, fmt.Errorf("write session manifest: %w", err)
+	}
+	return m.view(doc), nil
+}
+
+// checkSpace fails CLOSED. A probe error means no way to honour the floor, and
+// a refused session costs a retry while a wrong guess fills the volume.
+func (m *Manager) checkSpace(root string, need int64) error {
+	free, err := m.freeBytes(root)
+	if err != nil {
+		logger.Error("free-space probe failed; refusing upload", "root", root, "err", err)
+		return fmt.Errorf("%w: free-space probe failed: %v", ErrNoSpace, err)
```

golangci-lint (2.12.2)

[error] non-wrapping format verb for fmt.Errorf. Use `%w` to format errors

(errorlint)